app/views/users/index.html.erb

```diff
@@ -2,11 +2,21 @@
   <ul class="users-list">
     <% @users.each do |user| %>
       <li>
-        Name: <%= user.name %>
-        <span class="profile-link">
-          <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
-        </span>
+        <%= user.name %></br>
+          <%= link_to 'See Profile',  user_path(user), class: 'profile btn btn-outline-success btn-sm' %>
+        <% if signed_in? && !Friendship.reacted?(current_user.id, user.id) && current_user != user %>
+          <td> <%= link_to 'Add Friend', friendships_create_path(ids: {id1: current_user.id, id2: user.id}), class: 'friend btn btn-success btn-sm' %></td>
+        <% elsif signed_in? && !Friendship.confirmed_record?(current_user.id, user.id) && current_user != user %>
+          <% user.pending_friendships.each do |friendship| %>
+            <% if current_user %>
+              <%= link_to 'Cancel Request', friendships_destroy_path(friendship_id: friendship.id),  class: 'btn btn-danger btn-sm' %>
+              <% end %>
+          <% end %>
+        <% end %>
```

Comment:
That is what we call **logic in views**. I mean `if`, `if/else`, `if/elsif` and so on.
It should be implemented in **views helpers**